crates/compute/src/memory.rs

```diff
@@ -35,6 +35,9 @@ pub trait ComputeMemory<F> {
 	/// Borrows an immutable memory slice, narrowing the lifetime.
 	fn narrow<'a>(data: &'a Self::FSlice<'_>) -> Self::FSlice<'a>;
 
+	/// Borrows a mutable memory slice, narrowing the lifetime.
+	fn narrow_mut<'a, 'b: 'a>(data: Self::FSliceMut<'b>) -> Self::FSliceMut<'a>;
```

Comment:
Maybe it would be better to borrow instead of consuming the original value.
so that we can do the following:
```
let mut original_data = ...;
some_method(Mem::narrow(&mut original_data));
// continiue using original_data here